torch/_meta_registrations.py

```diff
@@ -655,7 +655,9 @@ def meta_adaptive_avg_pool2d(self, output_size):
         self.ndim == 3 or self.ndim == 4,
         lambda: f"Expected 3D or 4D tensor, but got {self.shape}",
     )
-    return self.new_empty(self.shape[:-2] + tuple(output_size))
+    output_shape = self.shape[:-2] + tuple(output_size)
+    memory_format = utils.suggest_memory_format(self)
+    return self.new_empty(output_shape).to(memory_format=memory_format)
```

Comment:
to(memory_format=) is a code smell from the discussion we had on Thursday.
What is it that you're trying to achieve here? You can use `self.new_empty_strided()` if you want to specify the full size/stride.

Reply:
Could you link me to that discussion ? why is it a code smell ?

Reply:
Because `.to(memory_format)` doesn't always do what you expect it too. It's fine here, but generally the results are a bit unpredictable, so better not use it at all #62027